src/soft-play.js

```diff
@@ -1,14 +1,58 @@
 // do not change these lines
 let adults = 0
 let children = 0
+let adultsTotal = 0
+let childrenTotal = 0
 
 // TODO: Write your functions in the below section. Your functions should update
 // the adults and children variables defined above.
 // Start with the occupancy function.
 
+//Enter function
+
+const enter = (numAdults, numChildren) => {
+  if ((numAdults === 0 && numChildren > 0) ||
+     (numChildren > numAdults)){
+    return false
+  }
+  adults += numAdults
+  children += numChildren
+
+  adultsTotal += numAdults
+  childrenTotal += numChildren
+ return true
+}
+
+//Leave function
+
+const leave = (numAdults, numChildren) => {
+  if  ((numAdults === 0) || 
+      ((adults - numAdults) < 0 || (children - numChildren) < 0) ||
+      ((adults - numAdults) < (children - numChildren)) ||
+      (numChildren > numAdults)){
+        return false
+  }
```

Comment:
Will you understand what this if statement is checking for in a couple of months?

I'd definitely move all of the checks into their own well named variables so you can have the code describe what it's doing a little better